test/core/compiler.jl

```diff
@@ -95,23 +88,26 @@ priors = 0 # See "new grammar" test.
         testmodel_comp(1.0, 1.2)
 
         # check if drawing from the prior works
-        @model testmodel0(x) = begin
+        @model testmodel0(x = missing) = begin
             x ~ Normal()
             return x
         end
         f0_mm = testmodel0()
         @test mean(f0_mm() for _ in 1:1000) ≈ 0. atol=0.1
 
         # Test #544
-        @model testmodel0(x = Vector{Float64}(undef, 2)) = begin
+        @model testmodel0(x = missing) = begin
+            if ismissing(x)
```

Comment:
What's the expected behaviour of "a user forget to write this condition while not providing `x` to the model"?

Reply:
`x` will be missing so `x[1]` will error. I think this is pretty intuitive.